Microsoft.Bot.Builder.Ai.LanguageGeneration/Resolver/LGResolver.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using DialogFoundation.Backend.LG;
+using LanguageGeneration.V2;
+using Microsoft.Bot.Builder.Ai.LanguageGeneration.Engine;
+using Microsoft.Bot.Builder.Ai.LanguageGeneration.Helpers;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Ai.LanguageGeneration.Resolver
+{
+    public class LGResolver : ILGResolver
+    {
+        private readonly ILGEndpoint _lgEndpoint;
+        private readonly ILGOptions _lgOptions;
+        private readonly IResolverPipelineFactory _resolverPipelineFactory;
+        private readonly IResolverPipeline _resolverPipeline;
+
+        public LGResolver(ILGEndpoint lgEndpoint, ILGOptions lgOptions)
+        {
+            _lgEndpoint = lgEndpoint ?? throw new ArgumentNullException(nameof(lgEndpoint));
+            _lgOptions = lgOptions ?? throw new ArgumentNullException(nameof(lgOptions));
+            _resolverPipelineFactory = new ResolverPipelineFactory();
+            _resolverPipeline = _resolverPipelineFactory.CreateResolverPipeline(_lgEndpoint.EndpointURI);
+        }
+
+        public async Task ResolveAsync(Activity activity, IDictionary<string, object> entities)
```

Comment:
Is there a reason to make the dictionary value object? Is it possible to have values other strings?

Reply:
yes, values could be one of 5 types